client/context/authContext.js

```diff
@@ -1,14 +1,38 @@
-import React, { useState, createContext } from 'react';
+import React, { useState, createContext, useEffect } from 'react';
+import { Redirect } from 'react-router-dom';
 
 const AuthContext = createContext();
 
 const AuthProvider = props => {
+    // localStorage.user
     const [auth, setAuth] = useState({
         token: localStorage.token || null,
-        user: null
+        user: JSON.parse(localStorage.getItem('user')) || null
     });
+    const isAuth = () => {
+        if (auth.user) {
+            return true;
+        }
+        return false;
+    };
+
+    const notGuest = prop_value => {
```

Comment:
Can we have a component like `OnlyGuestRoute` instead of putting this logic here ?